executorlib/task_scheduler/interactive/pysqaspawner.py

```diff
@@ -0,0 +1,214 @@
+import hashlib
+import os
+from time import sleep
+from typing import Callable, Optional
+
+from pysqa import QueueAdapter
+
+from executorlib.standalone.inputcheck import validate_number_of_cores
+from executorlib.standalone.interactive.spawner import BaseSpawner
+from executorlib.standalone.scheduler import pysqa_execute_command, terminate_with_pysqa
+from executorlib.task_scheduler.interactive.blockallocation import (
+    BlockAllocationTaskScheduler,
+)
+
+
+class PysqaSpawner(BaseSpawner):
+    def __init__(
+        self,
+        cwd: Optional[str] = None,
+        cores: int = 1,
+        threads_per_core: int = 1,
+        gpus_per_core: int = 0,
+        num_nodes: Optional[int] = None,
+        exclusive: bool = False,
+        openmpi_oversubscribe: bool = False,
+        slurm_cmd_args: Optional[list[str]] = None,
+        pmi_mode: Optional[str] = None,
+        config_directory: Optional[str] = None,
+        backend: Optional[str] = None,
+        **kwargs,
+    ):
+        """
+        Subprocess interface implementation.
+
+        Args:
+            cwd (str, optional): The current working directory. Defaults to None.
+            cores (int, optional): The number of cores to use. Defaults to 1.
+            threads_per_core (int, optional): The number of threads per core. Defaults to 1.
+            openmpi_oversubscribe (bool, optional): Whether to oversubscribe the cores. Defaults to False.
+        """
+        super().__init__(
+            cwd=cwd,
+            cores=cores,
+            openmpi_oversubscribe=openmpi_oversubscribe,
+        )
+        self._process: Optional[int] = None
+        self._threads_per_core = threads_per_core
+        self._gpus_per_core = gpus_per_core
+        self._num_nodes = num_nodes
+        self._exclusive = exclusive
+        self._slurm_cmd_args = slurm_cmd_args
+        self._pmi_mode = pmi_mode
+        self._config_directory = config_directory
+        self._backend = backend
+        self._pysqa_submission_kwargs = kwargs
+
+    def bootup(
+        self,
+        command_lst: list[str],
+    ):
+        """
+        Method to start the subprocess interface.
+
+        Args:
+            command_lst (list[str]): The command list to execute.
+        """
+        qa = QueueAdapter(
+            directory=self._config_directory,
+            queue_type=self._backend,
+            execute_command=pysqa_execute_command,
+        )
+        hash = hashlib.md5(str(self).encode()).hexdigest()
```

Comment:
At the moment the hash is used to generate a unique directory for each worker, but maybe it would be better to use the cache directory and the worker number.